frontend/src/lib/RunUtils.ts

```diff
@@ -67,6 +67,11 @@ function getFirstExperimentReference(run?: ApiRun | ApiJob): ApiResourceReferenc
   return run && getAllExperimentReferences(run)[0] || null;
 }
 
+function getFirstExperimentReferenceWithName(run?: ApiRun | ApiJob): ApiResourceReference | null {
+  return (run && getAllExperimentReferences(run) || [])
+    .find((ref) => !!ref.key && !!ref.key.id && !!ref.name) || null;
```

Comment:
Some nitpicking for typescript readability (from my previous experience, if there are project specific rules, please correct me):
* `(ref) => ref.key && ref.key.id && ref.name` is equivalent and simpler. No need to be super strict about returning a boolean.
* Returning a specific interface that is strict about nullity would make call code simpler, because there won't be a need to comment about why some fields are not null. Also, although this has only one usage now, you can see that a lot of other places can potentially use this instead too in the future.

Reply:
* Unfortunately, `ref.key && ref.key.id && ref.name` will actually return a `string | undefined` and results in a typing error for the `find` function that explicitly expects a `boolean`. I don't think a comment would be necessary here but it can be added to explain the necessity for explicit `boolean` conversion.
    * For Example:
         ```javascript
          "a" && "b" && "c" // Will return "c"
          "a" && "" && "c" // Will return ""
         ```
* I had an offline discussion with @IronPan about returning an interface and while we decided it would not be necessary, I agree that an interface would make the code call simpler. I'll make that change!

Reply:
Oops, you are right. I didn't realize it's actually a recent change of typescript to support this: [typescript playground example](http://www.typescriptlang.org/play/#code/MYewdgzgLgBAtgUwhAhgcwQLhtATgSzDRgF4YByACwQBsaQYB3EXGgE3IG4AoUSWFKRgBtMAFc6AGhgAGaWLBsEAM0II208igBGwcgF1e4aDG1CUAOlWKAFAA9SAPhh2AlEcggaCC-TQ3td25CKARcZRRgBBgAQQAHfAAlJDjjaIBvbhhsmBQAfmw8QjQeAF9uDxNcFLSIbHikmsgEYX0hYSyc9NLpdJRscj5QsFgQZVzycsM+EzQQEDZkiFTmoWrl2qtCNht1pxh1ixR3OYWllYhogDIrmBmvHz8bU8Wmy6PXIA)
If you switch to typescript 3.3, it shows the error as you pointed out. Our project is currently using 3.3.1.